lib/pubsub/topic.js

```diff
@@ -96,6 +102,37 @@ Topic.formatName_ = function(projectId, name) {
 };
 
 /**
+ * Wrapper for makeReq_ that automatically attempts to create a topic if it
+ * does not yet exist.
+ *
+ * @private
+ */
+Topic.prototype.autoCreateWrapper_ = function(method, path, q, body, callback) {
+  var self = this;
+
+  var createAndRetry = function(){
+    self.pubsub.createTopic(self.unformattedName, function(err) {
+      if (err) {
+        callback(err);
+        return;
+      }
+      self.origMakeReq_(method, path, q, body, callback);
+    });
+  };
+
+  this.origMakeReq_(method, path, q, body, function(err, res) {
+    if (method !== 'DELETE' && err && err.code === 404) {
```

Comment:
Ah, sorry I was referring to the outdated diff! It's already changed.
LGTM, but please wait for the maintainers' response.